server/src/main/java/org/candlepin/model/AsyncJobStatusCurator.java

```diff
@@ -86,4 +89,30 @@ public AsyncJobStatusCurator() {
 
         return this.getJobsInState(states);
     }
+
+    @Transactional
+    public AsyncJobStatus cancel(String jobId) {
+        this.cancelNoReturn(jobId);
+        AsyncJobStatus result = this.get(jobId);
+        if (result != null) {
+            this.refresh(result);
+        }
+        return result;
+    }
+
+    @Transactional
+    public void cancelNoReturn(String jobId) {
+        String hql = "update AsyncJobStatus j " +
+            "set j.state = :canceled " +
+            "where j.id = :jobid";
+        int updated = this.currentSession()
+            .createQuery(hql)
+            .setParameter("jobid", jobId)
+            .setInteger("canceled", JobState.CANCELED.ordinal())
+            .executeUpdate();
+        if (updated == 0) {
+            throw new NotFoundException("job not found");
```

Comment:
NotFoundException is for REST, not to be thrown from the curators.

Reply:
Done.